### PR TITLE
[WIP] Switch to TypeScript

### PR DESCRIPTION
**Pull request by theofidry**

### CODE REVIEW

File: src/Encoder/DecoderRegistry.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        } catch (error) {
+            return false;
+        }
+    }
+
+    /**
+     * @throws DecoderNotFound
+     * @private
+     */
+    _getDecoder(data: any, format: string, context: any[]): Decoder {
```
Comment (sgb-io) [inline src/Encoder/DecoderRegistry.ts:37]: i noticed you have a `private` `decoders` property on the class. but can we still not have private methods?

File: src/Encoder/Decoder.d.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * This file is part of the serializerjs package.
+ *
+ * (c) HAIRCVT <tfidry@haircvt.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+interface Decoder {
```
Comment (sgb-io) [inline src/Encoder/Decoder.d.ts:10]: 😃 yay!

File: src/Encoder/DecoderRegistry.ts (hunk 1 above)
Comment (theofidry) [inline src/Encoder/DecoderRegistry.ts:37]: They don't cf [SO question](http://stackoverflow.com/questions/16919473/private-functions-in-typescript) :/

Comment (theofidry) [inline src/Encoder/DecoderRegistry.ts:37]: More complete discussion [here](https://typescript.codeplex.com/discussions/448054)

**Pull request closed without merging**